packages/loader/container-loader/src/container.ts

```diff
@@ -598,16 +581,23 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         this.logger = ChildLogger.create(this.subLogger, "Container");
 
         this._deltaManager = this.createDeltaManager();
+        this.connectionStateHandler = new ConnectionStateHandler(
+            this,
+            this._deltaManager,
+            this.logger,
+            () => this._audience,
+            () => this.context,
+            () => this.protocolHandler);
 
         // keep track of last time page was visible for telemetry
         if (typeof document === "object" && document !== null) {
```

Comment:
This whole logic should be moved to the new class, and remove public Container.lastVisible

Reply:
No need for this now.